test/Scala/SbtDependencyTreeSpec.hs

```diff
@@ -0,0 +1,236 @@
+
+exampleMultiProjectSbtOut :: Text
+exampleMultiProjectSbtOut =
+  [r|[info] welcome to sbt 1.5.5 (Temurin Java 1.8.0_312)
+[info] Loading settings from plugins.sbt ...
+[info] Loading project definition from /som-path/project
+[info] Loading settings from build.sbt ...
+[info] Set current project to sbt-multi-example (in build file:/some-path/)
+[info] Formatting 1 Scala source in global:sbt ...
+[success] Total time: 1 s, completed 19-May-2022 2:38:08 PM
+[info] Updating {file:/some-path/}common...
+[info] Updating {file:/some-path/}global...
+[info] Done updating.
+[info] org.PARENT:PROJECT:1.0-SNAPSHOT [S]
+[info] Updating {file:/some-path/}some-project2...
+[info] Done updating.
+[info] Updating {file:/some-path/}some-project1...
+[info] org:PROJECTA:1.0-SNAPSHOT [S]
+[info]   +-org:B:1.0
+[info]   | +-org:C:1.0
+[info]   | +-org:D:1.0
+[info]   | 
+[info]   +-org:E:1.0
+[info]   +-org:F:1.0
+[info]     +-org:G:1.0
+[info]       +-org:H:1.0
+[info]       +-org:I:1.0
+[info]       | +-org:J:1.0
+[info]       | | +-org:K:1.0
+[info]       | | 
+[info]       | +-org:L:1.0
+[info]       | | +-org:M:1.0
+[info]       | | | +-org:N:1.0
+[info]       | | | +-org:O:1.0
+[info]       | | | 
+[info]       | | +-org:P:1.0
+[info]       | | +-org:Q:1.0
+[info]       | | 
+[info]       | +-org:R:1.0
+[info]       +-org:S:1.0
+[info]     
+[info] Done updating.
+[info] org:PROJECTB:1.0-SNAPSHOT [S]
+[info]   +-org:T:1.0
+[info]   | +-org:U:1.0
+[info]   | 
+[info]   +-org:W:1.0 (evicted by: 2.0)
+[info]     
+[info] Done updating.
+[info] 
+[success] Total time: 3 s, completed 19-May-2022 2:38:11 PM
+|]
+
+multiProjectGraphSpecForParent :: Graphing Dependency -> Spec
+multiProjectGraphSpecForParent graph = do
+  describe "buildGraph" $ do
+    it "should correctly graph dependencies" $ do
+      expectDirect [mkRawDep "org.PARENT:PROJECT" "1.0-SNAPSHOT"] graph
+
+multiProjectGraphSpecForProjectA :: Graphing Dependency -> Spec
+multiProjectGraphSpecForProjectA graph = do
+  let hasEdge :: Dependency -> Dependency -> Expectation
+      hasEdge = expectEdge graph
+
+  let hasDep :: Dependency -> Expectation
+      hasDep dep = expectDep dep graph
+
+  describe "buildGraph" $ do
+    it "should correctly graph dependencies" $ do
+      expectDirect [mkRawDep "org:PROJECTA" "1.0-SNAPSHOT"] graph
+
+      -- For sub project a
+      traverse_
+        hasDep
+        [ mkDep "B@1.0"
+        , mkDep "C@1.0"
+        , mkDep "D@1.0"
+        , mkDep "E@1.0"
+        , mkDep "F@1.0"
+        , mkDep "G@1.0"
+        , mkDep "H@1.0"
+        , mkDep "I@1.0"
+        , mkDep "J@1.0"
+        , mkDep "K@1.0"
+        , mkDep "L@1.0"
+        , mkDep "M@1.0"
+        , mkDep "N@1.0"
+        , mkDep "O@1.0"
+        , mkDep "P@1.0"
+        , mkDep "Q@1.0"
+        , mkDep "R@1.0"
+        , mkDep "S@1.0"
+        ]
+
+      hasEdge (mkDep "PROJECTA@1.0-SNAPSHOT") (mkDep "B@1.0")
```

Comment:
Is there a reason to use `hasEdge` here over `expectEdges`?

Reply:
Find it's easier to break down a component group for readability in tests (esp. when graph is somewhat long/complex)

Reply:
sounds good, thanks!